x-pack/plugins/ml/public/application/jobs/new_job/job_from_lens/create_job.ts

```diff
@@ -0,0 +1,283 @@
+
+      if (start === undefined || end === undefined || isNaN(start) || isNaN(end)) {
+        throw Error('Incompatible time range');
+      }
+    } catch (error) {
+      // eslint-disable-next-line no-console
+      console.error(error);
+      includeTimeRange = false;
+      start = undefined;
+      end = undefined;
+    }
+
+    // add job config and start and end dates to the
+    // job cloning stash, so they can be used
+    // by the new job wizards
+    stashJobForCloning(
+      {
+        jobConfig,
+        datafeedConfig,
+        createdBy,
+        start,
+        end,
+      } as JobCreatorType,
+      true,
+      includeTimeRange,
+      !includeTimeRange
+    );
+  } catch (error) {
+    // eslint-disable-next-line no-console
+    console.error(error);
+  }
+}
+
+export async function canCreateADJob(
+  vis: LensSavedObjectAttributes,
+  query: Query | undefined,
+  filters: Filter[] | undefined,
+  dataViewClient: DataViewsContract,
+  kibanaConfig: IUiSettingsClient
+) {
+  try {
+    const jobItems = await createADJobFromLensSavedObject(
+      vis,
+      query ?? getDefaultQuery(),
+      filters ?? [],
+      dataViewClient,
+      kibanaConfig
+    );
+
+    return !!jobItems;
+  } catch (error) {
+    return false;
+  }
+}
+
+async function createADJobFromLensSavedObject(
+  vis: LensSavedObjectAttributes,
+  query: Query,
+  filters: Filter[],
+  dataViewClient: DataViewsContract,
+  kibanaConfig: IUiSettingsClient
+) {
+  const { fields, timeField, splitField, dataView } = await extractFields(vis, dataViewClient);
+
+  const jobConfig = createEmptyJob();
+  const datafeedConfig = createEmptyDatafeed(dataView.title);
+
+  const mainQuery = query?.query !== '' ? query : vis.state.query;
+
+  const { combinedQuery } = createQueries(
+    { query: mainQuery, filter: filters },
+    dataView,
+    kibanaConfig
+  );
+  datafeedConfig.query = combinedQuery;
+
+  jobConfig.analysis_config.detectors = createDetectors(fields, splitField);
+
+  jobConfig.data_description.time_field = timeField.sourceField;
+  jobConfig.analysis_config.bucket_span = '15m';
+  if (splitField) {
+    jobConfig.analysis_config.influencers = [splitField.sourceField];
+  }
+
+  const createdBy =
+    splitField || jobConfig.analysis_config.detectors.length > 1
+      ? CREATED_BY_LABEL.MULTI_METRIC
+      : CREATED_BY_LABEL.SINGLE_METRIC;
+
+  return {
+    jobConfig,
+    datafeedConfig,
+    createdBy,
+  };
+}
+
+async function extractFields(vis: LensSavedObjectAttributes, dataViewClient: DataViewsContract) {
+  const visualization = vis.state.visualization as { layers: XYDataLayerConfig[] };
+  const indexpattern = vis.state.datasourceStates.indexpattern as IndexPatternPersistedState;
+
+  const compatibleLayers = visualization.layers.filter(
+    ({ layerType, seriesType }) =>
+      layerType === COMPATIBLE_LAYER_TYPE && COMPATIBLE_SERIES_TYPES.includes(seriesType)
+  );
+
+  const [firstCompatibleLayer] = compatibleLayers;
```

Comment:
If creating a job from a Lens with more than one compatible layer, can we show a message to indicate only the first compatible layer is being used?

Reply:
We could possibly show a toast.
Ideally we'd let the user pick the layer. This will be added when I add the flyout mentioned above.

Reply:
Is it worth adding a toast into this PR, or is the flyout definitely going to make 8.3?

Reply:
After discussing with @peteharverson we've agreed to push for adding the job creation flyout in 8.3.
So i'll defer adding this toast unless the flyout doesn't make it in.
I'll move on to creating the flyout as soon as this PR is merged.